models/ContainerPage.php

```diff
@@ -67,10 +69,84 @@ public function rules()
     {
         $rules = $this->defaultRules();
         $rules[] = ['in_new_window', 'integer'];
-        $rules[] = [['page_content'], 'safe'];
+	    $rules[] = [['page_content'], 'safe', 'when' => array($this, 'isNotHtml')];
+	    $rules[] = [['page_content'], 'filter', 'when' => array($this, 'isHtml'), 'filter' => array($this, 'purifyFilter')];
+
         return $rules;
     }
 
+	/**
+	 * @param ContainerPage $model the HTML code container
+	 * @return bool whether it is an html type container
+	 */
+	public function isHtml($model)
+	{
+		return $model->type == Container::TYPE_HTML;
+	}
+
+	/**
+	 * @param ContainerPage $model the HTML code
+	 * @return bool whether it is not an html type container
+	 */
+	public function isNotHtml($model)
+	{
+		return $model->type != Container::TYPE_HTML;
+	}
+
+	/**
+	 * Purify the HTML code only if its container type is html (conditional validation)
+	 *
+	 * @param string $html the HTML code to be purified
+	 * @return string the purified HTML code
+	 */
+    public function purifyFilter($html)
+    {
+	    return HtmlPurifier::process($html, $this->purifierConfig('test'));
```

Comment:
how are the presets selected? right now it is always using the `test` one?

Reply:
yep, just hardcoded, was thinking maybe add an option in the module settings, to select one of the preset, another proposition was to add the possibility to change directly the whitelist from within the module settings as a text field?

Reply:
Good job, but same question as @bumi , and why it's not set to 'full' ?

Reply:
Ok, I'll make it defaults to full preset.